notebooks/agent_powered_retrieval.ipynb

```diff
@@ -0,0 +1,1200 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #6.        top_k: Optional[int] = 5
</pre><p> </p><p><code>Optional[int]</code> means <code>Union[int, None]</code>.</p><p> </p><p>Since the function breaks if <code>top_k</code> is <code>None</code>,  I would use <code>top_k: int=5</code> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepset-ai/haystack-cookbook/pull/253/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>